File: Java/3117.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        // 입력 및 초기화
        Scanner scanner = new Scanner(System.in);
        Stack<Integer> stack = new Stack<>();

        int K, input, sum = 0;
        K = scanner.nextInt();

        // 계산부
        for(int i = 0; i < K; i++)
        {
            input = scanner.nextInt();

            // 입력받은 값이 0이 아니면 Push
            if(input != 0)
                stack.push(input);

                // 입력받은 값이 0면 Pop
            else
                stack.pop();
        }

        // 출력부
        System.out.println(stack.stream().reduce(0, Integer::sum));
    }
}

File: Python/3117.py
# K, 입력값 초기화
K, input_ = 0, 0
stack = []

# 입력부
K = int(input())

# 계산부
for i in range(K):
    input_ = int(input())

    if input_ == 0 : stack.pop()

    else : stack.append(input_)

# 출력부
print(sum(stack))